src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -379,6 +391,240 @@ void CascadiaSettings::_LoadDynamicProfiles()
+//   modify existing profiles or add new color schemes
+// - If the user settings has any namespaces in the "disabledProfileSources"
+//   property, we'll ensure that the corresponding folders do not get searched
+void CascadiaSettings::_LoadProtoExtensions()
+{
+    // First, accumulate the namespaces the user wants to ignore
+    std::unordered_set<std::wstring> ignoredNamespaces;
+    const auto disabledProfileSources = CascadiaSettings::_GetDisabledProfileSourcesJsonObject(_userSettings);
+    if (disabledProfileSources.isArray())
+    {
+        for (const auto& json : disabledProfileSources)
+        {
+            ignoredNamespaces.emplace(JsonUtils::GetValue<std::wstring>(json));
+        }
+    }
+
+    // Search through the local app data folder
+    if (std::filesystem::exists(wil::ExpandEnvironmentStringsW<std::wstring>(LocalAppDataFolder.data())))
+    {
+        _ApplyJsonStubsHelper(LocalAppDataFolder, ignoredNamespaces);
+    }
+
+    // Search through the global app data folder
+    if (std::filesystem::exists(wil::ExpandEnvironmentStringsW<std::wstring>(ProgramDataFolder.data())))
+    {
+        _ApplyJsonStubsHelper(ProgramDataFolder, ignoredNamespaces);
+    }
+
+    // Search through app extensions
+    // Gets the catalog of extensions with the name "com.microsoft.windows.terminal.settings"
+    const auto catalog = Windows::ApplicationModel::AppExtensions::AppExtensionCatalog::Open(winrt::to_hstring(AppExtensionHostName));
+
+    // Extracting the list of extensions from the catalog is an async operation, but since
+    // we are on a UI thread - and so cannot call a blocking operation like .get() - we use
+    // a mutex and condition variable
+    std::condition_variable cv;
+    std::mutex mtx;
+    Windows::Foundation::Collections::IVectorView<Windows::ApplicationModel::AppExtensions::AppExtension> extensions;
+
+    auto findAllExtensions = [&]() -> winrt::fire_and_forget {
+        co_await resume_background();
+        const auto localExtensions = catalog.FindAllAsync().get();
+
+        std::unique_lock<std::mutex> lock{ mtx };
+        extensions = localExtensions;
+        cv.notify_all();
+    };
+
+    std::unique_lock<std::mutex> lock{ mtx };
+    findAllExtensions();
+    cv.wait(lock);
+
+    for (const auto& ext : extensions)
+    {
+        // Only apply the stubs if the package name is not in ignored namespaces
+        if (ignoredNamespaces.find(ext.Package().DisplayName().c_str()) == ignoredNamespaces.end())
+        {
+            // Likewise, getting the public folder from an extension is an async operation
+            // So we use another mutex and condition variable
+            std::condition_variable cv2;
+            std::mutex mtx2;
+            Windows::Storage::StorageFolder foundFolder{ nullptr };
+
+            auto findPublicFolder = [&]() -> winrt::fire_and_forget {
+                co_await resume_background();
+                const auto localFolder = ext.GetPublicFolderAsync().get();

```

Comment:
using these APIs is the expensive bit. This is why I'm so worried. Not even the async part yet! Loading these DLLs takes time, which we actually got out of our critical path.

Would you mind profiling startup with an extension like this? :smile:

Reply:
This is what I got:

<img width="930" alt="profiling" src="https://user-images.githubusercontent.com/26824113/104071172-73e52d00-51bd-11eb-819f-bb63062f5bb4.png">